core/src/main/java/org/mule/util/xmlsecurity/XMLFactoryConfig.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.util.xmlsecurity;
+
+/**
+ * Cache key for XMLSecureFactoriesCache. Also creates the corresponding factory.
+ */
+public abstract class XMLFactoryConfig
+{
+    protected Boolean externalEntities;
```

Comment:
If they are private they would require getters to be accessed from the conflict classes that implement createFactory.